lib/galaxy/datatypes/genetics.py

```diff
@@ -827,6 +829,637 @@ def __init__(self, **kwd):
                                 substitute_name_with_metadata='base_name', is_binary=True)
 
 
+class LinkageStudies(Text):
+    """
+    superclass for classical linkage analysis suites
+    """
+    test_files = [
+        'linkstudies.allegro_fparam', 'linkstudies.allegro_ihaplo',
+        'linkstudies.alohomora_gts', 'linkstudies.alohomora_maf',
+        'linkstudies.alohomora_map', 'linkstudies.alohomora_ped',
+        'linkstudies.linkage_datain', 'linkstudies.linkage_map',
+        'linkstudies.linkage_pedin'
+    ]
+
+    def __init__(self, **kwd):
+        Text.__init__(self, **kwd)
+        self.lcount = 0
+        self.max_lines = 10
+        # iterate whole file without errors
+        self.eof_res = True
+
+    @staticmethod
+    def __is_binary_file(fstream):
+        """
+        Private binary file tester
+        """
+        result = False
+        if '\x00' in fstream.readline(512):
+            result = True
+
+        fstream.seek(0)
+        return result
+
+    @staticmethod
+    def tokenizer(line, sep=None):
+        """
+        General purpose string tokenizer
+        """
+        if sep is None:
+            return line.splitlines()[0].split(sep)
+
+        return line.splitlines()[0].split(sep)
+
+    def eof_function(self):
+        """
+        Overridable end-of-file function
+        """
+        return self.eof_res
+
+    def __per_line_op(self, line):
+        """
+        Private per-line operation and line counter
+        """
+        self.lcount += 1
+        if self.lcount > self.max_lines:
+            return -1
```

Comment:
Do you mean False here?

Reply:
no:
 - False would be passed to calling function which would then stop all further processing and reject the file
 - True would ..... and accept the file
 - None would tell the calling function that it needs to keep processing
 - -1 tells the calling function to stop further processing (max lines exceeded) and to evaluate the eof function

Reply:
using integers as return values by default seemed less readable, and I didn't want to implement needless enums